plugins/modules/keycloak_role_composites.py

```diff
@@ -0,0 +1,378 @@
+
+short_description: Allows administration of Keycloak composite roles with the Keycloak API
+
+version_added: 5.3.0
+
+description:
+    - This module allows you to add, remove or modify Keycloak roles to composite roles with the Keycloak REST API.
+      It requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+    - When updating composite role content, where possible provide the role ID to the module. This removes a lookup
+      to the API to translate the name into the role ID.
+
+
+options:
+    state:
+        description:
+            - State of the composite role.
+            - On C(present), specified composite role content will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), specified composite role content will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this role will reside.
+        default: 'master'
+
+    composite_role:
+        type: dict
+        description:
+            - Role representation of composite role
+            - This parameter is required
+        suboptions:
+            name:
+                type: str
+                description:
+                    - Name of the role.
+                    - This parameter is required only when creating or updating the role.
+            id:
+                type: str
+                description:
+                    - The unique identifier for this role.
+                    - This parameter is not required for updating or deleting a role but
+                      providing it will reduce the number of API calls required.
+
+    client_id:
+        type: str
+        description:
+            - Name of the client to be mapped (different than I(cid)).
+            - This parameter is required (can be replaced by cid for less API call).
+
+    cid:
+        type: str
+        description:
+            - Id of the client to be mapped.
+            - This parameter is not required for updating or deleting the rolemapping but
```

Comment:
There are some more occurrences of `rolemapping`. Are these intentional?

Reply:
I can rename it to role mapping.

Reply:
I'm not saying it has to be renamed, I'm mainly asking whether this is correct, since I don't know Keycloak.

Reply:
Oh, yes.